### PR TITLE
Update torch docs

### PR DESCRIPTION
**Pull request by mortid0**
- [x] Forked repo is synced with upstream -> github shows no code delta outside of the desired.
    https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork
- [x] Tests are passing? https://github.com/microsoft/DeepGNN/blob/main/CONTRIBUTING.md#run-tests
- [x] Changelog and documentation updated.
- [x] PR is labeled using the label menu on the right side.

Previous Behavior
----------------
Torch documentation suggested to use ray.datasets for distributed training.

New Behavior
----------------
1. Replace  ray.Dataset examples with torch.Dataprovider, because they are faster.
2. Add doc on how to work with ogb link prediction evaluation

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -5,6 +5,25 @@ load("@rules_python//python:defs.bzl", "py_binary", "py_test")
 load("@doc_deps//:requirements.bzl", doc_requirement = "requirement")
 load("@pip_deps//:requirements.bzl", "requirement")
 
+compatible_platforms = select({
+    "@platforms//os:osx": [],
+    "@platforms//os:linux": [],
```
Comment (coledie) [inline docs/BUILD:10]: Windows?

File: docs/BUILD [hunk 2]
```diff
@@ -272,114 +142,55 @@ py_test(
...
 
 py_test(
-    name = "torch_ray_usage",
+    name = "torch_distrib",
     srcs = ["doctest_template.py"],
-    args = ["torch/ray_usage.rst"],
+    args = ["torch/distrib.rst"],
     data = [
-        "torch/ray_usage.rst",
+        "torch/distrib.rst",
```
Comment (coledie) [inline docs/BUILD:163]: Rename tf/ray_usage similarly.

File: docs/advanced/meta_opt.rst [hunk 3]
```diff
@@ -1,252 +0,0 @@
-*****************************************
```
Comment (coledie) [inline docs/advanced/meta_opt.rst:1]: Why remove?

File: docs/torch/distrib.rst [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+.. code-block:: python
+
+    >>> from deepgnn.graph_engine.data.ppi import PPI
+    >>> ppi_data = PPI()
+
+
+GraphSage Model
+===============
+
+We'll reuse the GraphSage model from the link prediction example.
```
Comment (coledie) [inline docs/torch/distrib.rst:21]: Add a link

File: docs/torch/distrib.rst [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+    ...         )
+
+Experiment configuration
+========================
+Next step is to configure our experiment
+.. code-block:: python
+
+    >>> from deepgnn.graph_engine.snark.distributed import Server, Client as DistributedClient
+    >>> config = {
+    ...     "data_dir": "/tmp/ppi",  # graph.data_dir(),
```
Comment (coledie) [inline docs/torch/distrib.rst:158]: Why is graph.data_dir/num_nodes/... commented and we don't just use it?

File: docs/torch/distrib.rst [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+    >>> def make_predictions(model, loader):
+    ...     ns, xs, ys = [], [], []
+    ...     for batch in loader:
+    ...         nodes, node_features, cols, rows, node_label = (
+    ...             batch[6][0],
+    ...             batch[0][0],
+    ...             batch[2][0],
+    ...             batch[1][0],
+    ...             batch[5][0],
+    ...         )
```
Comment (coledie) [inline docs/torch/distrib.rst:225]: Might be nicer like
node_features, rows, cols, _, _, node_label, nodes = batch[:, 0]

File: docs/torch/distrib.rst [hunk 7]
```diff
@@ -0,0 +1,353 @@
...
+.. code-block:: python
+
+    >>> from deepgnn.graph_engine.snark.distributed import Server, Client as DistributedClient
+    >>> config = {
+    ...     "data_dir": "/tmp/ppi",  # graph.data_dir(),
+    ...     "device": torch.device("cpu"),
+    ...     "num_epochs": 2,
+    ...     "num_nodes": 56944,  # graph.NUM_NODES,
+    ...     "address": "localhost:9999",
+    ...     "feature_idx": 1,
```
Comment (coledie) [inline docs/torch/distrib.rst:163]: Also I thought we decided ray config wasn't great for examples and we should just hardcode the values instead?

File: docs/torch/distrib.rst (hunk 6 above)
Comment (coledie) [reply docs/torch/distrib.rst:225]: Same for above

File: docs/torch/distrib.rst [hunk 8]
```diff
@@ -0,0 +1,353 @@
...
+    ...     _, idx, counts = torch.unique(
+    ...         n, sorted=True, return_inverse=True, return_counts=True
+    ...     )
+    ...     _, ind_sorted = torch.sort(idx, stable=True)
+    ...     cum_sum = counts.cumsum(0)
+    ...     cum_sum = torch.cat((torch.tensor([0]), cum_sum[:-1]))
+    ...     first_indicies = ind_sorted[cum_sum]
+    ...
+    ...     y = torch.cat(ys, dim=0)
+    ...     x = torch.cat(xs, dim=0)
```
Comment (coledie) [inline docs/torch/distrib.rst:241]: NIT
    ...     x = torch.cat(xs, dim=0)
    ...     y = torch.cat(ys, dim=0)

File: docs/torch/distrib.rst [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+    ...             model,
+    ...             optimizer,
+    ...             train_dataloader,
+    ...         )
+    ...         loss /= train_dataset.count
+    ...         train_f1, test_f1 = eval_model(model, pred_train_dataloader, test_dataloader)
+    ...         get_logger().info(
+    ...             f"Epoch: {epoch:03d}, Loss: {loss:.4f}, Train F1: {train_f1:.4f}, Test F1: {test_f1:.4f}"
+    ...         )
+    ...     session.report({"test_f1": test_f1.item(), "train_f1": test_f1.item()})
```
Comment (coledie) [inline docs/torch/distrib.rst:340]: I think for multi-node usage, it is best we use session.report every epoch instead of get_logger().info since we don't want to check logs for each worker we just want the data all in one place with session.report?

File: docs/torch/link_pred.rst [hunk 10]
```diff
@@ -1,347 +1,402 @@
-****************************
-Custom Link Prediction Model
-****************************
+*******************************
+GraphSage Link Prediction Model
+*******************************
 
-In this guide we build a Link Prediction model to predict whether there is a link between any two nodes of the `caveman dataset <https://networkx.org/documentation/stable/reference/generated/networkx.generators.community.connected_caveman_graph.html?highlight=connected_caveman_graph#networkx.generators.community.connected_caveman_graph>`_.
+we'll take you step-by-step through the process of creating a link prediction model using GraphSage. This model will be trained on
```
Comment (coledie) [inline docs/torch/link_pred.rst:5]: NIT Capitalize

File: docs/torch/link_pred.rst [hunk 11]
```diff
@@ -1,347 +1,402 @@
...
-	>>> run_dist(
-	...     init_model_fn=create_model,
-	...     init_dataset_fn=create_dataset,
-	...     init_optimizer_fn=create_optimizer,
-	...     init_args_fn=init_args,
-	...     run_args=arg_list,
-	... )
+	>>> from ogb.linkproppred import Evaluator
+	>>> evaluator = Evaluator(name='ogbl-collab',)
+	>>> test(model, predictor, dataset.get_edge_split(), evaluator, config)
```
Comment (coledie) [inline docs/torch/link_pred.rst:401]: We should add some accuracy validation to ensure model keeps working as expected.

**Review by coledie — CHANGES_REQUESTED** [review]

File: docs/torch/distrib.rst (hunk 6 above)
Comment (mortid0) [reply docs/torch/distrib.rst:225]: Thanks!

File: docs/torch/distrib.rst (hunk 5 above)
Comment (mortid0) [reply docs/torch/distrib.rst]: Local testing artifact. Fixed.

File: docs/torch/distrib.rst (hunk 9 above)
Comment (mortid0) [reply docs/torch/distrib.rst]: Makes sense. Less code is also beter.

File: docs/torch/distrib.rst (hunk 7 above)
Comment (mortid0) [reply docs/torch/distrib.rst:163]: There will be way more hardcoded values involved here duplicating in multiple datasets we create on lines 271. I went back and forth about it. Current code looked a bit better. Neither is good though, so I am open to better solutions.

File: docs/advanced/meta_opt.rst (hunk 3 above)
Comment (coledie) [reply docs/advanced/meta_opt.rst:1]: I see, I think we should provide a link to Ray tune somewhere in our documentation so when users have a need, they can know the optimal solution.

**Review by coledie — APPROVED** [review]

File: docs/advanced/meta_opt.rst (hunk 3 above)
Comment (mortid0) [reply docs/advanced/meta_opt.rst:1]: I don’t see why do we need to do it. Ray has lots of other components. Why do we need to provide a link to this particular one?
